org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/models/SlicingModel.java

```diff
@@ -0,0 +1,134 @@
+package org.eclipse.scanning.api.device.models;
+
+import org.eclipse.scanning.api.ITimeoutable;
+import org.eclipse.scanning.api.annotation.ui.DeviceType;
+import org.eclipse.scanning.api.annotation.ui.FieldDescriptor;
+import org.eclipse.scanning.api.annotation.ui.FileType;
+
+public class SlicingModel implements ITimeoutable {
+	
+	@FieldDescriptor(minimum=1, hint="The rank of the data we will slice.")
+	private int dataRank = 2;
+
+	@FieldDescriptor(file=FileType.EXISTING_FILE)
+	private String   dataFile;
+	
+	/**
+	 * The name of the detector whose output we will be processing
+	 * 
+	 * This is used to figure out which part of the nexus file
+	 * to look at when processing.
+	 */
+	@FieldDescriptor(device=DeviceType.RUNNABLE, hint="The name of the detector whose output we will get.")
+	private String detectorName;
+	
+	@FieldDescriptor(editable=false, hint="The unique device name.")
+	private String name;
+	
+	@FieldDescriptor(editable=true, hint="The timeout of each data point during the scan in seconds.")
+	private long timeout = -1;
+
+	public SlicingModel() {

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1186)